services/monitor/monitor.sh

```
@@ -0,0 +1,368 @@
+# Checks for image updates and preserves previous version for rollback
+# Returns 0 if update available, 1 if current image is latest
+check_for_image_update() {
+    info "Checking for image updates"
+
+    local current_image_id=""
+    if container_exists; then
+        current_image_id=$(docker inspect "$CONTAINER_NAME" --format='{{.Image}}' 2>/dev/null || echo "")
+    fi
+
+    # Store current latest image ID before pulling - enables rollback if new version fails
+    local current_latest_id=$(docker inspect "$IMAGE_NAME" --format='{{.Id}}' 2>/dev/null || echo "")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)